extension/sumologicextension/extension.go

```diff
@@ -556,17 +556,17 @@ func (se *SumologicExtension) registerCollectorWithBackoff(ctx context.Context,
 	}
 }
 
-func (se *SumologicExtension) heartbeatLoop() {
+func (se *SumologicExtension) heartbeatLoop(ctx context.Context) {
 	if se.registrationInfo.CollectorCredentialID == "" || se.registrationInfo.CollectorCredentialKey == "" {
 		se.logger.Error("Collector not registered, cannot send heartbeat")
 		return
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	go func() {
-		// When the close channel is closed ...
-		<-se.closeChan
+		// When the context is done ...
+		<-ctx.Done()
 		// ... cancel the ongoing heartbeat request.
 		cancel()
```

Comment:
I meant to remove `defer cancel`. Essentially what I'm trying to do here is pass the parent context into the heartbeat loop so that when the extension stops the heartbeats stop immediately.